cpp/src/arrow/util/iterator.h

```diff
@@ -63,30 +64,26 @@ class Iterator : public util::EqualityComparable<Iterator<T>> {
   explicit Iterator(Wrapped has_next)
       : ptr_(new Wrapped(std::move(has_next)), Delete<Wrapped>), next_(Next<Wrapped>) {}
 
-  Iterator() : ptr_(NULLPTR, NoopDelete) {}
+  Iterator() : ptr_(NULLPTR, [](void*) {}) {}
```

Comment:
Why not keep `NoopDelete`?

Reply:
It doesn't do enough to keep an explicit function